### ISSUE TITLE
Error with successful conversion: "type" as required property missing?

### ISSUE THREAD

**Issue by chrisgrieser** [manubot/manubot#299]
When running pandoc with the `pandoc-manubot-cite` filter, I get the following error message. However, the pandoc conversion is sucessful and identifiers are all converted properly. (I am trying to run `pandoc-manubot-cite` as a filter independently from the rest of manubot. I am not entirely sure whether this is the cause of this problem?)

```
## WARNING'type' is a required propertyrequired element missing at: 0## WARNING'type' is a required propertyrequired element missing at: 0## WARNING'type' is a required propertyrequired element missing at: 0 
```

for reference, my pandoc command with all options
```
pandoc '/Users/chrisgrieser/Google Drive/File Hub/test.md' -o '/Users/chrisgrieser/Google Drive/File Hub/test 15-08-2021.docx' --filter='pandoc-manubot-cite' --citeproc --resource-path='/Users/chrisgrieser/Google Drive/File Hub/' --resource-path='/Users/chrisgrieser/Google Drive/File Hub/attachments/' --bibliography '/Users/chrisgrieser/Google Drive/BibDesk/My Library.bib' --csl '/Users/chrisgrieser/Google Drive/Dokumente/IT Backups/Alfred.alfredpreferences/workflows/user.workflow.41B90DCD-A99E-4943-A19A-E91859557FB0/citation-styles/research-policy.csl' --number-sections --from=markdown+lists_without_preceding_blankline --metadata link-citations=true --metadata date="$(date "+%e. %B %Y")" --reference-doc '/Users/chrisgrieser/Google Drive/Academia/Vorlagen/Academic Writing Template.docx'
```
```
pandoc 2.14.1
Compiled with pandoc-types 1.22, texmath 0.12.3, skylighting 0.11,
citeproc 0.4.1, ipynb 0.1.0.1

manubot version v0.5.1
```

**Comment by agitter** [comment]
That error message is new to me.  Hopefully the warning status and your successful conversion indicate it may be safe to ignore.

A few quick Google and GitHub issue searches didn't reveal the source of that warning.  I don't believe it comes directly from the pandoc-manubot-cite filter.

Could you please attach your `test.md` file in this issue to help us debug further?  We'll see whether we can reproduce the issue from that input file alone or whether all the input files are needed.

**Comment by dhimmel** [comment]
Note that this is a poorly constructed WARNING message and not an ERROR. It occurs when a CSL JSON record contains an invalid `type` ([list of valid types here](https://github.com/citation-style-language/schema/blob/c2142118a0265dfcf7d66aa3328251bedcc66af2/schemas/input/csl-data.json#L12-L56)). See https://github.com/manubot/manubot/issues/112#issue-447400151.

The warning is coming from

https://github.com/manubot/manubot/blob/c27e5b607a5724b21d198bcfea72c2687d09a262/manubot/cite/citeproc.py#L129-L134

Would be nice to fix this so the warning message is more precise and doesn't occur multiple times.

**Comment by chrisgrieser** [comment]
Ah I see, I got it.
My `test.md` document is simply
```
# Introduction

Lorem ipsum [@doi:10.15363/thinklab.4].
```
So no issues there. But based on @dhimmel's explanation, I tried the following two commands:

```
pandoc test.md -o test3.docx --filter=pandoc-manubot-cite --citeproc
pandoc test.md -o test3.docx --filter=pandoc-manubot-cite --citeproc --bibliography "/Users/chrisgrieser/Google Drive/BibDesk/My Library.bib"
```
And the first one runs without any issue, while the latter produces the same error, even though the bibliography file isn't even actually used.

I looked into my bibliography file and it indeed includes some types like "incollection" (i.e. bookchapters) that are not in the list linked by @dhimmel. That the warning occurs multiple times (I only copypasted a part of it) is likely caused by the number of entries with "invalid" types I guess?

---

So the good news seems to be that this isn't a major issue. So I guess I can use manubot only as a pandoc-filter and just have to ignore the warning message until its fixed in the next release?

However, I am wondering whether such a warning message should occur at all – running pandoc with citeproc does work on my bibtex file without any issues. And, at least in my usecase, the pandoc-manubot-filter does not even need to actually read the .bib file, since citeproc handles this.

**Comment by dhimmel** [comment]
> That the warning occurs multiple times (I only copypasted a part of it) is likely caused by the number of entries with "invalid" types I guess?

Not necessarily. The warning occurs as part of the CSL JSON validation and pruning. In this process, the CSL JSON is passed through the pruner 5 times:

https://github.com/manubot/manubot/blob/c27e5b607a5724b21d198bcfea72c2687d09a262/manubot/cite/citeproc.py#L30

> just have to ignore the warning message until its fixed in the next release?

Yes, ignore it for now until we fix it. The fix would likely be a better warning message that doesn't repeat itself or suppressing it altogether.

> pandoc-manubot-filter does not even need to actually read the .bib file, since citeproc handles this.

With the `pandoc-manubot-filter`, the manubot filter does need to read the bib file since it needs to know which references to generate metadata for and which to take from the bib file.

Sounds like with your use case, you are not able to update the bib file to be clean? Since it's a file imported from elsewhere?

**Comment by chrisgrieser** [comment]
I see. thanks for the explanation!

> Sounds like with your use case, you are not able to update the bib file to be clean? Since it's a file imported from elsewhere?

I use a bibtex-file created by [BibDesk](https://bibdesk.sourceforge.io/). But I believe the entry type `incollection` wasn't exclusive to BibDesk, but also used e.g. by [Zotero's BetterBibTeX Plugin](https://retorque.re/zotero-better-bibtex/).
